src/store/modules/topics.store.js

```diff
@@ -32,56 +30,53 @@ const state = {
 
 const getters = {
     isDefaultTopic: (state) => {
-        return state.current && state.current.id === 'ech'
+        return state.current === 'ech'
     },
     /** Returns the current topic's id, or `ech` if no topic is selected */
     currentTopicId: (state) => {
-        if (state.current) {
-            return state.current.id
-        }
-        return 'ech'
+        return state.current
+    },
+    currentTopic: (state) => {
+        return state.config.find((topic) => topic.id === state.current)
     },
 }
 
 const actions = {
-    setTopics: ({ commit }, topics) => {
-        if (Array.isArray(topics)) {
-            commit('setTopicConfig', topics)
-        }
-    },
-    setTopicTree: ({ commit }, tree) => {
-        if (Array.isArray(tree)) {
-            commit('setTopicTree', tree)
-        }
+    setTopics: ({ commit }, { topics, dispatcher }) => {
+        commit('setTopics', { topics, dispatcher })
     },
-    changeTopic: ({ commit }, topic) => {
-        commit(CHANGE_TOPIC_MUTATION, topic)
+    setTopicTree: ({ commit }, { layers, dispatcher }) => {
+        commit('setTopicTree', { layers: layers.map((layer) => layer.clone()), dispatcher })
     },
-    setTopicById: ({ commit, state }, topicId) => {
-        const topic = state.config.find((topic) => topic.id === topicId)
-        if (topic) {
-            commit(CHANGE_TOPIC_MUTATION, topic)
+    changeTopic: ({ commit, state }, { value, dispatcher }) => {
+        if (
+            state.config.find((topic) => topic.id === value) ||
+            // during appLoadingManagement.routerPlugin the topics are not yet set
+            // therefore we cannot validate the topic ID
+            dispatcher === 'appLoadingManagement.routerPlugin'
+        ) {
+            commit('changeTopic', { value, dispatcher })
         } else {
-            log.error('No topic found with ID', topicId)
+            log.error(`Invalid topic ID ${value}`)
         }
     },
-    setTopicTreeOpenedThemesIds: ({ commit }, themes) => {
-        if (typeof themes === 'string') {
-            commit(
-                'setTopicTreeOpenedThemesIds',
-                themes.indexOf(',') !== -1 ? themes.split(',') : [themes]
-            )
-        } else if (Array.isArray(themes)) {
-            commit('setTopicTreeOpenedThemesIds', themes)
+    setTopicTreeOpenedThemesIds: ({ commit }, { value, dispatcher }) => {
+        if (typeof value === 'string') {
+            commit('setTopicTreeOpenedThemesIds', {
+                themes: value.indexOf(',') !== -1 ? value.split(',') : [value],
```

Comment:
`.split` always returns an array (even if no match is found) so we could simplify this by doing
`themes: values.split(',')` (it's already 100% sure this is a string with the if above)